src/generator/writers/writeFunction.ts

```diff
@@ -614,6 +636,20 @@ function writeCondition(
     returns: TypeRef | null | string,
     ctx: WriterContext,
 ) {
+    const threwCode = extractThrowErrorCode(f.trueStatements);
+    const isAloneIf =
+        f.falseStatements === undefined || f.falseStatements.length === 0;
+
+    if (!elseif && isAloneIf && threwCode !== undefined) {
+        // if (cond) { throw(X) } => throw_if(X, cond)
+        // if (!cond) { throw(X) } => throw_unless(X, cond)
```

Comment:
This is semantically non-equivalent change in the presence of mutation. For instance,

```tact
extends mutates fun inc(self: Int): Int { self += 1; return self }
contract Test {
    receive() {  }
    // returns 43
    get fun ifThrow(): Int {
        let x: Int = 42;
        if (x.inc() == 0) {
            throw(x.inc());
        }
        return x;
    }
    // returns 44
    get fun throwIf(): Int {
        let x: Int = 42;
        throwIf(x.inc(), x.inc() == 0);
        return x;
    }
}
```

Reply:
Indeed, fixed